jwst/resample/resample_spec_step.py

```diff
@@ -38,53 +37,87 @@ def process(self, input):
-                pixfrac=self.pixfrac, kernel=self.kernel,
-                fillval=self.fillval, good_bits=self.good_bits)
-            # Do the resampling
-            resamp.do_drizzle()
-
-            for model in resamp.output_models:
+            # result is a DrizProductModel
+            result = self._process_slit(input_models)
+        return result
+
+    def _process_multislit(self, input_models):
+        """
+        Resample MultiSlit data
+
+        Parameters
+        ----------
+        input : `~jwst.datamodels.ModelContainer`
+            A container of `~jwst.datamodels.MultiSlitModel`
+
+        Returns
+        -------
+        result : `~jwst.datamodels.MultiProductModel`
+            The resampled output, one per source
+        """
+        containers = multislit_to_container(input_models)
+        result = datamodels.MultiProductModel()
+        result.update(input_models[0])
+        for container in containers.values():
+            resamp = resample_spec.ResampleSpecData(container, **self.drizpars)
+            drizzled_models = resamp.do_drizzle()
+
+            for model in drizzled_models:
                 model.meta.cal_step.resample = "COMPLETE"
                 model.meta.asn.pool_name = input_models.meta.pool_name
                 model.meta.asn.table_name = input_models.meta.table_name
+                update_s_region_spectral(model)
 
-            # Return either the single resampled datamodel, or the container
-            # of datamodels.
-            if len(resamp.output_models) == 1:
-                result = resamp.output_models[0]
+            # Everything resampled to single output model
+            if len(drizzled_models) == 1:
+                result.products.append(drizzled_models[0])
+                result.products[-1].bunit_data = container[0].meta.bunit_data
             else:
-                result = resamp.output_models
+                # For singlely-resampled images
+                for model in drizzled_models:
+                    result.products.append(model)
+                    result.products[-1].bunit_data = container[0].meta.bunit_data
+
+        return result
 
-        for model in result.products:
+    def _process_slit(self, input_models):
+        """
+        Resample Slit data
+
+        Parameters
+        ----------
+        input : `~jwst.datamodels.ModelContainer`
+            A container of `~jwst.datamodels.ImageModel`
+            or `~jwst.datamodels.SlitModel`
+
+        Returns
+        -------
+        result : `~jwst.datamodels.DrizProductModel`
+            The resampled output, one per source
+        """
+        resamp = resample_spec.ResampleSpecData(input_models, **self.drizpars)
+        drizzled_models = resamp.do_drizzle()
+
+        for model in drizzled_models:
+            model.meta.cal_step.resample = "COMPLETE"
+            model.meta.asn.pool_name = input_models.meta.pool_name
+            model.meta.asn.table_name = input_models.meta.table_name
```

Comment:
If ``pool_name`` and ``table_name`` are located in the ``asn`` section of model.meta in the output model, why do they come from the top level of meta in the input models?

Reply:
`input_models` is a ModelContainer, which stores these in `meta` because it currently still has a schema.  Removing the schema from ModelContainer is on my TODO list, and that will require these lines of code to be changed.  I think the solution is to just put any data that a container needs to carry around with in a class attribute or property.  Attributes could be just created on the fly as needed as well, as there elsewhere in the code.  👀